src/aioswitcher/api/messages.py

```diff
@@ -130,16 +131,22 @@ def get_thermostat_remote_id(self) -> str:
         remote_hex = unhexlify(self._hex_response)
         return remote_hex[84:92].decode().rstrip("\x00")
 
-    def get_shutter_position(self) -> int:
-        """Return the current shutter position."""
-        hex_pos = self._hex_response[152:154].decode()
+    def get_shutter_position(self, index: int) -> int:
+        """Return the current position of the shutter 0 <= pos <= 100."""
+        index -= 1
```

Comment:
I see that the `index -= 1` is repeating in multiple places here, wouldn't it be better to use zero based indexing? after all this is something internal and we programmers preferer counting from zero.

Reply:
Done.

Reply:
> Done.

Where exactly is it done?

Reply:
https://github.com/TomerFi/aioswitcher/pull/767/files#diff-76fda1abe206ffc78578e59e694e42cf998f9158ec5427d72e3f42d8db31c7a4R226
Here I can see the change..

Reply:
Oh this is something else.. let me see

Reply:
In this case, I can't change this part..
This is part of the logic..
If the index is 1 then it should be 0 if this is 2 then it should be 1 and so on..
This is how this works..

Reply:
Everywhere in this PR you reduce 1 from the index (about 6 occasion), than the only place you can get the index is a method you implanted which can return 1, 3 or 0 (which was returning -1 that I also don't understand why it is now changed to 0 and what is it used for).
Why can't this method return 0 or 2 so you don't have to reduce it by one later everywhere?

> This is how this works..

That is not how PR review works, there is a method that return -1 (changed to 0 without any explanation why), 3 or 1 without any documentation what it means, you have no plan to explain why you wrote it like that, I don't mind we resolve this conversation and I let someone else review it.

Reply:
I will try to explain..
The changes here: https://github.com/TomerFi/aioswitcher/pull/767/commits/630a921d800c2fdd81306cf031b9f4e5397b104c
I changed based on your request as here it does not matter if this is -1 or 0 because if the conditions are not applied it does not matter what is returning.

The other index -= 1 are parts of this logic to return the correct packet and it should remain..

Reply:
I think we are having a broken discussion. I didn't ask to return 0 instead of -1 from `get_shutter_index`, if this condition can't be met you should either return `None` or raise.

I asked that we use zero based index instead of having to do `index -= 1` on all of the methods here. I don't see this breaking any method as instead of passing 3 to a method and let it subtract 1 to get 2, you can just pass 2.
I fully understand from the code here how the packets are generated and I don't understand your comment:
> The other index -= 1 are parts of this logic to return the correct packet and it should remain..

So please explain it better if I am missing something.

Reply:
Change has done..